components/script/dom/htmlimageelement.rs

```diff
@@ -1010,6 +1017,36 @@ pub fn collect_sequence_characters<F>(s: &str, predicate: F) -> (&str, &str)
     return (s, "");
 }
 
+//https://html.spec.whatwg.org/multipage/#parse-a-sizes-attribute
+pub fn parse_a_sizes_attribute(value: DOMString, width: Option<u32>) -> SourceSizeList {
+    let mut input = ParserInput::new(&value);
+    let mut parser = Parser::new(&mut input);
+    let url = ServoUrl::parse("about:blank").unwrap();
+    let context = ParserContext::new(
+        Origin::Author,
+        &url,
+        Some(CssRuleType::Style),
+        ParsingMode::empty(),
+        QuirksMode::NoQuirks,
+        None,
+    );
+    let mut source_size_list = SourceSizeList::parse(&context, &mut parser);
+    if source_size_list.source_sizes.len() == 0  {
+        source_size_list = match width {
+            Some(w) => SourceSizeList {
+                source_sizes: vec![],
+                value: Some(Length::from_px(w as f32))
+            },
+            None => SourceSizeList {
+                source_sizes: vec![],
+                value: Some(Length::NoCalc(NoCalcLength::ViewportPercentage(ViewportPercentageLength::Vw(100.))))
```

Comment:
Also note that this is not needed, `SourceSizeList::evaluate` already handles this correctly.